frontend/app/controllers/skill-search.js

```diff
@@ -14,51 +13,138 @@ export default class SkillSearchController extends Controller {
   @service
   router;
 
-  @tracked
-  levelValue = this.level || 1;
-
-  @tracked
-  interestValue = this.interest || 1;
-
-  currentSkillId = this.skill_id;
-
-  @tracked
-  skills = this.store.findAll("skill", { reload: true });
+  @action
+  initFilters() {
+    let filters = [];
+    let skill_id = new URLSearchParams(window.location.search).get("skill_id");
+    let level = new URLSearchParams(window.location.search).get("level");
+    let interest = new URLSearchParams(window.location.search).get("interest");
+    if (skill_id && level && interest) {
+      let skill_ids = skill_id.split(",");
+      let levels = level.split(",");
+      let interests = interest.split(",");
+      for (let i = 0; i < skill_ids.length; i++) {
+        filters.pushObject({
+          selectedSkill: this.store.findRecord("skill", parseInt(skill_ids[i])),
+          currentSkillId: skill_ids[i],
+          levelValue: levels[i],
+          interestValue: interests[i]
+        });
+      }
+    } else {
+      filters = [
+        {
+          selectedSkill: null,
+          currentSkillId: null,
+          levelValue: 1,
+          interestValue: 1
+        }
+      ];
+    }
+    this.set("filters", filters);
+  }
 
-  @computed("model", "skills")
-  get selectedSkill() {
-    const skillId = this.skill_id;
-    return skillId ? this.store.peekRecord("skill", skillId) : null;
+  @computed
+  get skills() {
+    return this.store.findAll("skill", { reload: true });
   }
 
   updateSelection() {
+    let skill_ids = "",
+      levels = "",
+      interests = "";
+    for (let i = 0; i < this.get("filters").length; i++) {
+      if (this.get("filters." + i + ".currentSkillId") !== null) {
```

Comment:
Without the null comparison, ember tries to get the new skill list every time you add a new filter, this results in a failing request, since the filter hasn't any option selected at that point